### PR TITLE
Gora-444 : Add #size() to Result API

### PR DESCRIPTION
**Pull request by carlosrmng**
This PR adds  #size() to Result API, it implements size() for existing backends. The complete disscusion can be found in [GORA-444](https://issues.apache.org/jira/browse/GORA-444).

Some issues with this PR:
-new tests are need.
-Backends: Accumulo, Avro, HBase, don't support size() functions, so an approach using batchSize was used instead.

Code review welcome, thanks

### CODE REVIEW

**Comment by lewismc** [comment]
@carlosrmng great 👍 
Can you please format 2 space indents the same as the other source code?
Thanks

**Comment by lewismc** [comment]
I am +1 for committing this. @carlosrmng it would be great if we could add tests to [DataStoreTestBase.java](https://github.com/apache/gora/blob/master/gora-core/src/test/java/org/apache/gora/store/DataStoreTestBase.java) to accommodate this proposed addition to the core API.

**Comment by nishadi** [comment]
This is great work...!!! As Lewis mentioned, shall we add tests for this? As well we will need to document all the public APIs as well.

**Comment by lewismc** [comment]
Yep tests would be great. IMHO, as this is something we want to test for every Datastore, I suggest it be implemented within [DataStoreTestUtil.java](https://github.com/apache/gora/blob/master/gora-core/src/test/java/org/apache/gora/store/DataStoreTestUtil.java)

**Comment by djkevincr** [comment]
+1

**Comment by carlosrmng** [comment]
Thanks for your support, I will start working in some tests and documentation.

File: gora-jcache/src/main/java/org/apache/gora/jcache/query/JCacheResult.java [hunk 1]
```diff
@@ -82,4 +82,9 @@ protected boolean nextInner() throws IOException {
     return true;
   }
 
+  @Override
+  public int size() {
+    return cacheKeySet.size();
+  }
+
```
Comment (nishadi) [inline gora-jcache/src/main/java/org/apache/gora/jcache/query/JCacheResult.java]: @carlosrmng It is a good practice if you can avoid empty lines like this :)

File: gora-core/src/main/java/org/apache/gora/memory/store/MemStore.java [hunk 2]
```diff
@@ -92,7 +92,9 @@ public boolean nextInner() throws IOException {
 
     @Override
     public int size() {
-      return map.navigableKeySet().size();
+      int totalSize = map.navigableKeySet().size();
+      int intLimit = (int)this.limit;
+      return intLimit > 0 && totalSize>intLimit ? intLimit : totalSize;
```
Comment (nishadi) [inline gora-core/src/main/java/org/apache/gora/memory/store/MemStore.java]: Please reformat the code with desired spacing.

**Comment by nishadi** [comment]
Hi Carlos, This is great. One thing to note is when sending PRs please make sure code is well formatted, all required Java docs is added for public methods. These improvements will show case a good standard in your coding :)

**Comment by cguzel** [comment]
+1

**Comment by lewismc** [comment]
+1

File: gora-jcache/src/main/java/org/apache/gora/jcache/query/JCacheResult.java [hunk 3]
```diff
@@ -88,4 +88,4 @@ public int size() {
     int intLimit = (int)this.limit;
     return intLimit > 0 && totalSize>intLimit ? intLimit : totalSize;
   }
-}
+}
```
Comment (nishadi) [inline gora-jcache/src/main/java/org/apache/gora/jcache/query/JCacheResult.java]: Please add the ending  new line

**Comment by carlosrmng** [comment]
Is this okay for merge?

**Comment by djkevincr** [comment]
If any one have concerns please raise, otherwise I will proceed merging this PR to master.

**Comment by djkevincr** [comment]
Awesome work Carlos :+1: One minor thing to point out, it s good if you can be consistent with commit messages. Have look at this post. 
[1] https://chris.beams.io/posts/git-commit/

**Pull request merged**